microk8s-resources/actions/common/utils.sh

```diff
@@ -73,6 +73,33 @@ use_manifest() {
     rm "${tmp_manifest}"
 }
 
+addon_name() {
+    # Extracts the addon from the argument.
+    # addons can have arguments in the form of <addon-name>:<arg1>=<value1>;<arg2>=<value2>
+    # Example: enable linkerd:proxy-auto-inject=on;other-args=xyz
+    # Parameter:
+    #   $1 the full addon command
+    # Returns:
+    #   <addon-name>
+
+    local IFS=':'
+    read -ra ADD_ON <<< "$1"
+    echo "${ADD_ON[0]}"
+}
+
+addon_arguments() {
+    # Extracts the addon arguments.
+    # Example: enable linkerd:proxy-auto-inject=on;other-args=xyz
+    # Parameter:
+    #   $1 the addon arguments in array
+    # Returns:
+    #   add-on arguments array
+    local IFS=':'
+    read -ra ADD_ON <<< "$1"
+    local IFS=';'
+    read -ra ARGUMENTS <<< "${ADD_ON[1]}"
```

Comment:
This does not work for more than one arguments, see for example:
```
$ microk8s.enable linkerd:arg1;arg2
Enabling Linkerd2
Enabling DNS
Applying manifest
service/kube-dns unchanged
serviceaccount/kube-dns unchanged
configmap/kube-dns unchanged
deployment.extensions/kube-dns configured
Restarting kubelet
DNS is enabled
namespace/linkerd unchanged
serviceaccount/linkerd-controller unchanged
clusterrole.rbac.authorization.k8s.io/linkerd-linkerd-controller unchanged
clusterrolebinding.rbac.authorization.k8s.io/linkerd-linkerd-controller unchanged
serviceaccount/linkerd-prometheus unchanged
clusterrole.rbac.authorization.k8s.io/linkerd-linkerd-prometheus unchanged
clusterrolebinding.rbac.authorization.k8s.io/linkerd-linkerd-prometheus unchanged
service/linkerd-controller-api unchanged
service/linkerd-proxy-api unchanged
deployment.extensions/linkerd-controller configured
customresourcedefinition.apiextensions.k8s.io/serviceprofiles.linkerd.io unchanged
serviceaccount/linkerd-web unchanged
service/linkerd-web unchanged
deployment.extensions/linkerd-web configured
service/linkerd-prometheus unchanged
deployment.extensions/linkerd-prometheus configured
configmap/linkerd-prometheus-config unchanged
serviceaccount/linkerd-grafana unchanged
service/linkerd-grafana unchanged
deployment.extensions/linkerd-grafana configured
configmap/linkerd-grafana-config unchanged
Linkerd is starting

Command 'arg2' not found, did you mean:

  command 'args' from deb ucommon-utils

Try: sudo apt install <deb name>


```

Reply:
Yes this one, the issue here is that the shell treated ; as different commands.  For this kind of scenario it has to be enclosed in quotes.